extension/src/popup/SorobanContext.tsx

```diff
@@ -26,19 +28,51 @@ export const SorobanProvider = ({
   pubKey: string;
 }) => {
   const networkDetails = useSelector(settingsNetworkDetailsSelector);
-  const { sorobanRpcUrl = "" } = networkDetails;
 
-  const server = new SorobanRpc.Server(sorobanRpcUrl, {
-    allowHttp: sorobanRpcUrl.startsWith("http://"),
-  });
+  let server: SorobanContextInterface["server"];
+  let newTxBuilder: SorobanContextInterface["newTxBuilder"];
+  if (!networkDetails.sorobanRpcUrl) {
+    // handle any issues with a network missing sorobanRpcUrl
+    let serverUrl;
+
+    switch (networkDetails.network) {
+      case NETWORKS.FUTURENET:
+        serverUrl = SOROBAN_RPC_URLS[NETWORKS.FUTURENET];
+        break;
+      case NETWORKS.TESTNET:
+        serverUrl = SOROBAN_RPC_URLS[NETWORKS.TESTNET];
+        break;
+      case NETWORKS.PUBLIC:
+        serverUrl = SOROBAN_RPC_URLS[NETWORKS.PUBLIC];
+        break;
+      default:
+        serverUrl = SOROBAN_RPC_URLS[NETWORKS.PUBLIC];
```

Comment:
this is a bit heavy handed, but we'll be refactoring this whole thing into a helper, so this is a band-aid to unblock users who get stuck

Reply:
using a switch because indexing SOROBAN_RPC_URLS with an enum was kind of messy

Reply:
sounds good! Wondering if we should still throw in the default case since it will error downstream anyway?

Reply:
yeah that's a good idea. Using the mainnet rpc with the wrong network will lead to weird user experience